include/generic/SugarWidgets/SugarWidgetSubPanelDetailViewLink.php

```diff
@@ -121,9 +121,17 @@ public function displayList(&$layout_def)
             if ($module == 'EAPM') {
                 $link = "index.php?module=$module&action=$action&record={$record}{$parent}";
             }
-            return '<a href="' . $link . '" >'."$value</a>";
-        } else {
-            return $value;
-        }
-    }
+            if( $module == 'DocumentRevisions'){
+            	$view_revision = 
+            		'&nbsp;<a href="index.php?preview=yes&entryPoint=download&id=' . 
+            			$record . '&type=Documents" class="tabDetailViewDFLink" target="_blank" style="border-bottom: 0px;"><i class="glyphicon glyphicon-eye-open"></i>';
```

Comment:
`$record` value came from the layout defs but it's never validated. The layout def sturcture could be wrong for e.g `$layout_def['target_record_key']` and `$layout_def['fields']['ID']` both are unset, or `$layout_def['fields'][strtoupper($layout_def['target_record_key'])]` is not set, the `$record` getting a null value which is an invalid record ID. Please check the valid layout_def structure at the top of the function and/or validate the record ID here. The error should be in a log note warning.